wp_api/src/request.rs

```diff
@@ -279,6 +262,34 @@ impl Debug for WpNetworkRequest {
     }
 }
 
+/// Unique identifier for this request – used for external record-keeping
+#[uniffi::export(with_foreign)]
+pub trait NetworkRequestDescriptor: Send + Sync {
+    fn request_id(&self) -> String;
+    fn method(&self) -> RequestMethod;
+    fn url(&self) -> WpEndpointUrl;
+    fn header_map(&self) -> Arc<WpNetworkHeaderMap>;
```

Comment:
These don't seem related to the purpose for the trait?

Reply:
@jkmassel Could you explain what you mean?

To give some context, @crazytonyli asked to add more methods to this trait so that he can generalize some things in Swift between `WpNetworkRequest` & `MediaUploadRequest`.

Reply:
Right, so `NetworkRequestDescriptor` seems to indicate "a way to describe the request".

Using it to "generalize some things in Swift between WpNetworkRequest & MediaUploadRequest" is a different thing though?

I wonder if the name is still correct in this case?

Reply:
I don't love the name either, but couldn't find a suggestion to improve yet 😅 

Our goal is to describe a request type without a body. Open for suggestions?

Reply:
As discussed on our call today, I've renamed it as `NetworkRequestAccessor`. We both agreed that this is still not the greatest name, but until we find something better we'll go with it.